### ISSUE TITLE
Develop a JSON structure for seed state

### ISSUE THREAD

**Issue by sg4e** [sg4e/MaikaTracker#44]
Here's an idea:
```json
{
    "version": "1.0",
    "flags": "bAAMI_akOAKgp41g6Lg",
    "seed": "RELISHRACE",
    "keyItems": [
        {
            "name": "Magma Key",
            "collected": false
        },
        {
            "name": "Hook",
            "location": "Baron Inn",
            "collected": true
        }
    ],
    "completedChecks": ["Start", "Antlion", "Fabul", "Mt. Ordeals", "Baron Inn"]
}
```
The full structure of the state would contain a lot more fields but something like this is the starting point.

There are open questions about the form of the fields and how they're best designed. For example, `"completedChecks"` could instead be implemented as a list of all viable logic checks of the flagset, each with a `name` field and a `completed` boolean.

The usefulness of a structured way of storing tracked state of a seed goes beyond MaikaTracker. After defining a structure for these json state files, the schema can be used in other FF4FE tools. Possibilities include Router's Relish route-optimization tools and "seed replay" tools where a player would be able to visualize their progression through the seed via their routing choices.

**Comment by CaitSith2** [comment]
Full json nearly implemented.

    {
        "version" : "1.0",
        "text_flags" : "V1 Jisa Kqmt Pk Cx -nodupes -rescue -startcecil -hobs T2gr S4q -nosirens B F Nck Gwl Etf Xsbk -spoon -fab -huh -z",
        "binary_flags" : "bAAMI_akOAKgp41g6Lg",
        "seed" : "RELISHRACE",
        "keyItems" : [
            {
                "name" : "CRYSTAL",
                "location" : "KOKKOL"
            },
            {
                "name" : "PASS",
                "location" : "BARON_INN",
                "collected" : true
            },
            {
                "name" : "HOOK",
                "location" : "ANTLION"
            },
            {
                "name" : "DARKNESS",
                "location" : "LEVIATAN"
            },
            {
                "name" : "EARTH",
                "location" : "DLUNAR"
            },
            {
                "name" : "TWIN_HARP",
                "location" : "WYVERN"
            },
            {
                "name" : "PACKAGE",
                "location" : "S30"
            },
            {
                "name" : "SAND_RUBY",
                "location" : "ZOT"
            },
            {
                "name" : "BARON_KEY",
                "location" : "G5"
            },
            {
                "name" : "MAGMA_KEY",
                "location" : "E22",
                "collected" : true
            },
            {
                "name" : "TOWER_KEY",
                "location" : "RAT_TAIL"
            },
            {
                "name" : "LUCA_KEY",
                "location" : "SHEILA_PANLESS"
            },
            {
                "name" : "ADAMANT",
                "location" : "OGOPOGO"
            },
            {
                "name" : "LEGEND",
                "location" : "PALE_DIM",
                "collected" : true,
                "used" : true
            },
            {
                "name" : "PAN",
                "location" : "DWARF_CASTLE",
                "collected" : true
            },
            {
                "name" : "SPOON",
                "location" : "U6"
            },
            {
                "name" : "RAT_TAIL",
                "location" : "L13"
            },
            {
                "name" : "PINK_TAIL",
                "location" : "BAHAMUT"
            }
        ],
        "locationsVisited" : [
            "BARON_INN",
            "FABUL",
            "ASURA",
            "PALE_DIM",
            "ORDEALS",
            "DWARF_CASTLE"
        ],
        "bosses" : [
            {
                "name" : "MistD",
                "location" : "Rubicante"
            },
            {
                "name" : "Soldier",
                "location" : "Kainazzo"
            },
            {
                "name" : "Octo",
                "location" : "Leviath"
            },
            {
                "name" : "Antlion",
                "location" : "Yang"
            },
            {
                "name" : "WHag",
                "location" : "CPU"
            },
            {
                "name" : "Mombomb",
                "location" : "Ogopogo",
                "seen" : true
            },
            {
                "name" : "Gauntlet",
                "location" : "Plague"
            },
            {
                "name" : "Milon",
                "location" : "Eblan"
            },
            {
                "name" : "MilonZ",
                "location" : "MistD"
            },
            {
                "name" : "DKCecil",
                "location" : "Guards"
            },
            {
                "name" : "Guards",
                "location" : "Baigan"
            },
            {
                "name" : "Yang",
                "location" : "DarkImps"
            },
            {
                "name" : "Baigan",
                "location" : "PaleDim"
            },
            {
                "name" : "Kainazzo",
                "location" : "Milon"
            },
            {
                "name" : "DElf",
                "location" : "Wyvern"
            },
            {
                "name" : "MagusSis",
                "location" : "DElf",
                "seen" : true,
                "defeated" : true
            },
            {
                "name" : "Valvalis",
                "location" : "Fiends"
            },
            {
                "name" : "Calcabrina",
                "location" : "Asura"
            },
            {
                "name" : "Golbez",
                "location" : "MilonZ"
            },
            {
                "name" : "Lugae",
                "location" : "EvilWall"
            },
            {
                "name" : "DarkImps",
                "location" : "Octo"
            },
            {
                "name" : "Eblan",
                "location" : "MagusSis"
            },
            {
                "name" : "Rubicante",
                "location" : "DKCecil"
            },
            {
                "name" : "EvilWall",
                "location" : "LunarD"
            },
            {
                "name" : "Fiends",
                "location" : "Valvalis"
            },
            {
                "name" : "CPU",
                "location" : "Bahamut"
            },
            {
                "name" : "Odin",
                "location" : "WHag"
            },
            {
                "name" : "Asura",
                "location" : "Golbez"
            },
            {
                "name" : "Leviath",
                "location" : "Odin"
            },
            {
                "name" : "Bahamut",
                "location" : "Calcabrina"
            },
            {
                "name" : "PaleDim",
                "location" : "Mombomb"
            },
            {
                "name" : "LunarD",
                "location" : "Antlion"
            },
            {
                "name" : "Plague",
                "location" : "Soldier"
            },
            {
                "name" : "Ogopogo",
                "location" : "Lugae"
            },
            {
                "name" : "Wyvern",
                "location" : "Gauntlet"
            }
        ],
        "characters" : [
            "PALADIN_CECIL",
            null,
            "RYDIA",
            null,
            "KAIN"
        ],
        "openedChests" : [
            "Z1",
            "Z3",
            "Z5",
            "E22",
            "B1",
            "B2",
            "S1",
            "S2",
            "S3",
            "S9",
            "S10",
            "G4"
        ],
        "shopItems" : {
            "Troia[Pub]" : [
                "Starveil",
                "Kamikaze"
            ],
            "Feymarch" : [
                "Cabin",
                "Exit"
            ],
            "DwarfCastle" : [
                "Bacchus",
                "Hourglass1"
            ],
            "Silvera" : [
                "Ether2",
                "Moonveil"
            ],
            "EblanCave" : [
                "Coffin",
                "Stardust"
            ],
            "Fabul" : [
                "Cure2",
                "Cure3",
                "Ether1",
                "Elixir"
            ],
            "Tomara" : [
                "Hourglass2"
            ],
            "Agart" : [
                "Grimoire"
            ],
            "Mysidia" : [
                "Illusion",
                "AgApple"
            ],
            "Hummingway" : [
                "Cure1",
                "Silkweb"
            ],
            "Kaipo" : [
                "SomaDrop"
            ],
            "Baron" : [
                "Life",
                "Tent",
                "GaiaDrum",
                "AuApple",
                "MuteBell"
            ],
            "Troia[Item]" : [
                "Hourglass3"
            ]
        }
    }

**Comment by CaitSith2** [comment]
a23fb4e4bf93261a3733058a011a63634c3dc832 - Code to save/load JSON now live in #43

**Comment by sg4e** [comment]
Implemented in #43
